cmd/cluster-display/main.go

```diff
@@ -0,0 +1,200 @@
+	}
+	return o, nil
+}
+
+func validateOptions(o options) error {
+	_, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %w", err)
+	}
+	if o.hiveKubeconfigPath == "" {
+		return fmt.Errorf("--hive-kubeconfig must be set")
+	}
+	return nil
+}
+
+type options struct {
+	logLevel           string
+	port               int
+	gracePeriod        time.Duration
+	hiveKubeconfigPath string
+}
+
+func addSchemes() error {
+	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
+		return fmt.Errorf("failed to add hivev1 to scheme: %w", err)
+	}
+	return nil
+}
+
+func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client) (*Page, error) {
+	clusterImageSetMap := map[string]string{}
+	clusterImageSets := &hivev1.ClusterImageSetList{}
+	if err := hiveClient.List(ctx, clusterImageSets); err != nil {
+		return nil, fmt.Errorf("failed to list cluster image sets: %w", err)
+	}
+	for _, i := range clusterImageSets.Items {
+		clusterImageSetMap[i.Name] = i.Spec.ReleaseImage
+	}
+
+	clusterPools := &hivev1.ClusterPoolList{}
+	if err := hiveClient.List(ctx, clusterPools); err != nil {
+		return nil, fmt.Errorf("failed to list cluster pools: %w", err)
+	}
+
+	page := Page{Data: []map[string]string{}}
+	for _, p := range clusterPools.Items {
+		maxSize := "nil"
+		if p.Spec.MaxSize != nil {
+			maxSize = strconv.FormatInt(int64(*p.Spec.MaxSize), 10)
+		}
+		releaseImage := clusterImageSetMap[p.Spec.ImageSetRef.Name]
+		owner := p.Labels["owner"]
+		page.Data = append(
+			page.Data, map[string]string{
+				"namespace":    p.Namespace,
+				"name":         p.Name,
+				"ready":        strconv.FormatInt(int64(p.Status.Ready), 10),
+				"size":         strconv.FormatInt(int64(p.Spec.Size), 10),
+				"maxSize":      maxSize,
+				"imageSet":     p.Spec.ImageSetRef.Name,
+				"labels":       labels.FormatLabels(p.Labels),
+				"releaseImage": releaseImage,
+				"owner":        owner,
+			},
+		)
+	}
+	return &page, nil
+}
+
+func getResource(ctx context.Context, hiveClient ctrlruntimeclient.Client, resource string) (*Page, error) {
+	switch strings.ToLower(resource) {
+	case "clusterpools":
+		return getClusterPoolPage(ctx, hiveClient)
+	default:
+		return nil, fmt.Errorf("unknown resource: %s", resource)
+	}
+}
+
+func getRouter(ctx context.Context, hiveClient ctrlruntimeclient.Client) *http.ServeMux {
+	handler := http.NewServeMux()
+
+	handler.HandleFunc("/api/health", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		page := map[string]bool{"ok": true}
+		if err := json.NewEncoder(w).Encode(page); err != nil {
+			logrus.WithError(err).WithField("page", page).Error("failed to encode page")
+		}
+	})
+
+	handler.HandleFunc("/api/v1/clusterpools", func(w http.ResponseWriter, r *http.Request) {
+		logrus.WithField("path", "/api/v1/clusterpools").Info("serving")
+		page, err := getResource(ctx, hiveClient, "clusterpools")
```

Comment:
Please get rid of the `getResource` function and just call `getClusterPoolPage` directly

Reply:
Updated. Sorry that I did not get this point before.